### ISSUE TITLE
Problems with IHealthChecksBuilder.AddTypeActivatedCheck<T> in  Microsoft.Extensions.DependencyInjection.HealthChecksBuilderAddCheckExtensions

### ISSUE THREAD

**Issue by RyanLiu99** [dotnet/AspNetCore.Docs#25707]
AddTypeActivatedCheck<T> support additional arguments to concert type  constructor which implements IHealthCheck.
This sounds good idea, but it brought few issues. 

First, I feel it is not really needed. 

More important, it might cause memory leak. 

It resolves (some) HealthCheck constructor arguments from IoC container, and not resolve HealthCheck itself from Ioc container ( to support additional constructor arguments).    This bring few issues:

The runtime create particular HealthCheck instance each time Health end point is hit.  Since itself is not managed by Ioc container, lifetime does not apply to it. Ioc container cannot dispose it even it is a IDisposable. Heath check Runtime just create it but does not dispose it either.   Application cannot dispose it since it was not created by application itself. 

So it was created each time when health check is performed and never disposed. That can cause resource/memory leak. 
I call it memory leak introduced by Ioc container leak --  HealthCheck instances are not managed by Ioc container, but their dependencies are.  Those dependencies are referred by HealthCheck  but their lifetime can out of sync, this can cause another type issue. 

---
#### Document Details

⚠ *Do not edit this section. It is required for docs.microsoft.com ➟ GitHub issue linking.*

* ID: c0cbb094-29e9-740a-9fd6-db372641d753
* Version Independent ID: 52d8efc9-a336-71d9-6d90-98173cbf6bf4
* Content: [Health checks in ASP.NET Core](https://docs.microsoft.com/en-us/aspnet/core/host-and-deploy/health-checks?view=aspnetcore-6.0)
* Content Source: [aspnetcore/host-and-deploy/health-checks.md](https://github.com/dotnet/AspNetCore.Docs/blob/main/aspnetcore/host-and-deploy/health-checks.md)
* Product: **aspnet-core**
* Technology: **aspnetcore-hostdeploy**
* GitHub Login: @Rick-Anderson
* Microsoft Alias: **riande**

**Comment by Rick-Anderson** [comment]
@rafikiassumani-msft  please assign a reviewer.

**Comment by rafikiassumani-msft** [comment]
@halter73  Please review.

**Comment by halter73** [comment]
I do not think we need to complicate our sample `IHealthCheck` implementation.

> AddTypeActivatedCheck support additional arguments to concert type constructor which implements IHealthCheck.
This sounds good idea, but it brought few issues.
>
> First, I feel it is not really needed.

@RyanLiu99, you might not need to pass constructor arguments, and I admit it's uncommon, but I was able to find at least [one usage](https://github.com/Daniel15/dnstools/blob/f57626d68a6d9e61eae4d7ee8866c773f0f9e168/src/DnsTools.Web/Startup.cs#L105).

> More important, it might cause memory leak.

If you expected an `IHealthCheck` with an `IDisposable` implementation to get disposed, I can see it. No where do we indicate that the `IHealthCheck` will get disposed. All we say is that it's "type activated".

@RyanLiu99 would something like

> NOTICE: The [IHealthCheck](https://learn.microsoft.com/en-us/dotnet/api/microsoft.extensions.diagnostics.healthchecks.ihealthcheck?view=dotnet-plat-ext-6.0) implementation will not be disposed unless it is resolved from [RequestServices](https://learn.microsoft.com/en-us/dotnet/api/microsoft.aspnetcore.http.httpcontext.requestservices?view=aspnetcore-6.0). [AddTypeActivatedCheck](https://learn.microsoft.com/en-us/dotnet/api/microsoft.extensions.dependencyinjection.healthchecksbuilderaddcheckextensions.addtypeactivatedcheck?view=dotnet-plat-ext-6.0#microsoft-extensions-dependencyinjection-healthchecksbuilderaddcheckextensions-addtypeactivatedcheck-1(microsoft-extensions-dependencyinjection-ihealthchecksbuilder-system-string-system-object())) always activates the healthcheck type with [CreateInstance<T>(IServiceProvider, Object[])](https://learn.microsoft.com/en-us/dotnet/api/microsoft.extensions.dependencyinjection.activatorutilities.createinstance?view=dotnet-plat-ext-6.0#microsoft-extensions-dependencyinjection-activatorutilities-createinstance-1(system-iserviceprovider-system-object())) without attempting to resolve it from `RequestServices` in order to support arguments.
